### PR TITLE
[5.7] Handle long tagged NSStrings

### PR DESCRIPTION
**Pull request by Catfish-Man**
Cherry-pick of https://github.com/apple/swift/pull/60428

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

File: stdlib/public/core/SmallString.swift [hunk 1]
```diff
@@ -346,13 +346,22 @@ extension _SmallString {
   //
   @_effects(readonly) // @opaque
   @usableFromInline // testable
-  internal init(taggedCocoa cocoa: AnyObject) {
+  internal init?(taggedCocoa cocoa: AnyObject) {
```
Comment (stephentyrone) [inline stdlib/public/core/SmallString.swift:349]: Are we certain that this isn't ever used from an inlinable public API?

Comment (Catfish-Man) [reply stdlib/public/core/SmallString.swift:349]: We are. It *is*, however, used from test code, which I had to fix up.

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
There we go, that got the tests sorted. Good to go now.

**Review by stephentyrone — APPROVED** [review]

**Pull request merged**